backend/tool_instance/tool_instance_helper.py

```diff
@@ -442,11 +444,17 @@ def validate_tool_access(
         user: User,
         tool_uid: str,
     ) -> None:
-        prompt_regitry_tool = PromptStudioRegistry.objects.get(pk=tool_uid)
+        # HACK: Assume tool_uid is a prompt studio exported tool and query it.
+        # We suppress ValidationError when tool_uid is of a static tool.
+        try:
+            prompt_registry_tool = PromptStudioRegistry.objects.get(pk=tool_uid)
+        except DjangoValidationError:
+            logger.info(f"Not validating tool access for tool: {tool_uid}")
+            return
```

Comment:
@chandrasekharan-zipstack A doubt here, are we expecting only `DjangoValidationError` to be thrown from the try statement ?

Reply:
Probably a `PromptStudioRegistry.DoesNotExist` exception might also be raised when the object is not present. Ideally, it'll be good to handle such cases generically. I didn't deep dive into improving these parts in the PR since I just wanted a quick fix. We can review and improve the code as we face issues

Reply:
Understood, should we add a generic catch here to pick that up ? Or a to-do to revisit later.

Ideally these issues will not pop-up is what I am assuming from your message, then it's okay to continue without these too.

Reply:
@athul-rs as much as possible we should avoid the habit of using a generic catch - what's ideal is catching something specific and handling it with a user friendly message. All uncaught exceptions would be bubbled to Django which will then raise a generic 500 error. So the user will only see `Something went wrong` but our logs would have the entire traceback which will allow us to fix it

Reply:
Yeah, makes sense.